Description: Link the LLDBDebugger plugin with the LLDBProtocol libary first
 Make LLDDebugger link against LLDBProtocol before lldb itself so the lldb
 library doesn't get dropped when using -Wl,--as-needed. 
Author: James Cowgill <james410@cowgill.org.uk>
Forwarded: no
Last-Update: 2014-06-06
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/LLDBDebugger/CMakeLists.txt
+++ b/LLDBDebugger/CMakeLists.txt
@@ -80,8 +80,8 @@
         # Define the output - shared library
         add_library(${PLUGIN_NAME} SHARED ${PLUGIN_SRCS})
 
-        target_link_libraries(${PLUGIN_NAME} ${LIBLLDB})
         target_link_libraries(LLDBDebugger LLDBProtocol)
+        target_link_libraries(${PLUGIN_NAME} ${LIBLLDB})
 
         # Codelite plugins doesn't use the "lib" prefix.
         set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
